src/lib/curve-cache.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+const BigNumber = require('bignumber.js')
+const PrefixMap = require('ilp-routing').PrefixMap
+
+/**
+ * Cache liquidity curve quotes so that they can be referenced during payment.
+ */
+class CurveCache {
+  /**
+   * @param {Object} params
+   * @param {Integer} params.quoteExpiry milliseconds
+   * @param {Integer} params.quoteCleanupInterval milliseconds
+   */
+  constructor (params) {
+    this.sources = {} // { sourceLedger ⇒ { appliesToPrefix ⇒ [liquidityQuote] } }
+    this.quoteExpiryDuration = params.quoteExpiry
+    this.quoteCleanupInterval = params.quoteCleanupInterval
+  }
+
+  start () {
+    setInterval(this.removeExpiredQuotes.bind(this), this.quoteCleanupInterval)
+  }
+
+  removeExpiredQuotes () {
```

Comment:
Why not just leave expired curves in the cache, and check their expiry date when you retrieve it?